lego/apps/users/models.py

```diff
@@ -522,3 +531,48 @@ def ignore_date(date):
         elif winter_to < (date.month, date.day) < winter_from:
             return False
         return True
+
+
+class PhotoConsent(BasisModel):
+    user = models.ForeignKey(
+        User, related_name="photo_consents", on_delete=models.CASCADE
+    )
+    semester = models.CharField(max_length=6)
+    year = models.PositiveIntegerField()
+    domain = models.CharField(choices=constants.PHOTO_CONSENT_DOMAINS, max_length=100)
+    is_consenting = models.BooleanField(blank=True, null=True, default=None)
+
+    def get_consents(self, user):
+        photo_consents = PhotoConsent.objects.filter(user=user)
+        now = timezone.now()
+        current_semester = "autumn" if now.month > 7 else "spring"
+        current_year = now.year
+        if not any(
+            pc.year == current_year
+            and pc.semester == current_semester
+            and pc.domain == "SOCIAL_MEDIA"
+            for pc in photo_consents
```

Comment:
This probably works, but is very slow and unnecessary. Do all this as the DB level by creating a proper filter query and use `<Queryset>.exists()` instead of the for loop.